Fix vfprintf_chk and add other xprintf_chk functions

thomasten opened the pull request:
* Added missing parameter to __vfprintf_chk
* Added other ..printf_chk functions similar to the two existing ones

dthaler commented on libc/__fprintf_chk.c:
```diff
@@ -0,0 +1,23 @@
+#include <stdarg.h>
+#include <stdio.h>
+
+// GCC sometimes replaces fprintf() calls with __fprintf_chk() calls. In glibc
```
should this code be inside
#ifdef __GNUC__
?

radhikaj commented:
@thomasten. Thank you for this contribution. Can you please address feedback. We would love to be able to merge this after it gets approved.

thomasten commented:
> 
> 
> @thomasten. Thank you for this contribution. Can you please address feedback. We would love to be able to merge this after it gets approved.

@radhikaj I think I already addressed all feedback in 1b95835

jhand2 commented on libc/__fprintf_chk.c:
```diff
@@ -0,0 +1,27 @@
+#include <stdarg.h>
+#include <stdio.h>
+
+// GCC sometimes replaces fprintf() calls with __fprintf_chk() calls. In glibc
```
What version of GCC is this and in what situations are you seeing these calls generated? If code is compiled with `-nostdlib`, calls to code in glibc shouldn't be linked, right?

thomasten replied:
I think it happened when adding a static library to my enclave. I guess this is not supposed to be supported and one should recompile such dependencies?

jhand2 replied:
I could go either way. OE provides a standard C library, so I don't know that dependencies should all need to be compiled with nostdlib (or some other combination of compiler arguments). But supporting every internal function for every compiler could get a bit out of hand. I guess if someone wanted to target a different libc implementation in a non-enclave application they would have to recompile the dependencies in the same way.

@anakrish @CodeMonkeyLeet do either of you have an opinion here?

anakrish replied:
GCC emits calls to these object size checking functions when compiled with _FORTIFY_SOURCE and optimizations https://gcc.gnu.org/onlinedocs/gcc/Object-Size-Checking.html

Secure coding practices recommend compiling with _FORTIFY_SOURCE (and also other flags like -fstack-protector) https://wiki.debian.org/Hardening

See https://github.com/openenclave/openenclave/issues/1600 for a list of secure flags that OE must be built with.

jhand2 replied:
Ok thats fine. For reference, BSD libc just aliases the functions (i.e. forwards __fprintf_chk to fprintf).

So I'm fine with supporting this in OE.

radhikaj commented:
@thomasten can you squash the commits? I can then merge this?

radhikaj commented:
Bors r+

jhand2 commented:
Test failure looks like an infra issue unrelated to this PR

bors r+
